runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -98,6 +98,127 @@ function M.on_refresh(err, _, ctx, _)
   return vim.NIL
 end
 
+--- @class vim.lsp.inlay_hint.get.opts
+--- @field bufnr integer?
+--- @field filter function?
+--- @field id integer?
+--- @field name string?
+--- @field range lsp.Range?
+---
+--- @class vim.lsp.inlay_hint.get.resp
+--- @field bufnr integer
+--- @field client_id integer
+--- @field inlay_hint lsp.InlayHint
+
+--- Get the list of inlay hints, (optionally) restricted by buffer, client, or range.
+--- This method does not make an LSP request, but just queries locally stored inlay hints.
+--- The returned inlay hints are unresolved.
+---
+--- Example usage:
+---
+--- ```lua
+--- local hint = vim.lsp.inlay_hint.get({ bufnr = bufnr })[1]
+--- resolved_hint = client.request('inlayHint/resolve', hint.inlay_hint, callback, bufnr)
+--- vim.lsp.util.apply_text_edits(resolved_hint.textEdits, bufnr, 'utf-8')
+--- command = resolved_hint.label[1].command
+--- vim.lsp.buf.execute_command({ command = command })
+--- ```
```

Comment:
Docs and some example code about [inlayHint/resolve](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#inlayHint_resolve) is indeed great to have here! As a suggestion, I wish this example usage could be a self-contained code that can run without much modification and edits needed. Or even easier to understand.

Line 121 (`client.request`) looks like a synchronous call, so I guess you probably meant `request_sync` (or rewrite the remaining lines in a callback/handler).
If meant an asynchronous `client.request`, `callback` is not defined in this context so it's not very clear here. Another nitpicking is that we seem to prefer calling it as `handler` rather than a `callback`?

Reply:
I should've tested the example, my bad. It should be working and complete now